### PR TITLE
Feat/stats

### PR DESCRIPTION
**Pull request by montanizstills**
Please take a look at the //TODO inside `PlayerStats.java`

### CODE REVIEW

File: src/main/java/com/github/obc/webserver/models/PlayerStats.java [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+private String PTS;
+private String RAT;
+private String REB;
+private String SC_EFF;
+private String SH_EFF;
+private String STL;
+private String STL_TO;
+private String TD3;
+private String TECH;
+private String TO;
```
Comment (ghost1497) [inline src/main/java/com/github/obc/webserver/models/PlayerStats.java:40]: Change to camel case clear variable names (i.e. fieldGoalPercent, averagePointsPerGame, etc.)

File: src/main/java/com/github/obc/webserver/models/PlayerStats.java [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    private String pointsPerGame;
+    private String rating;
+    private String reboundsPerGame;
+    private String scoringEfficiency;
+    private String shootingEfficiency;
+    private String stealsPerGame;
+    private String stealTurnoverRatio;
+    private String tripleDoubles;
+    private String technicalFouls;
+    private String turnoversPerGame;
```
Comment (ghost1497) [inline src/main/java/com/github/obc/webserver/models/PlayerStats.java:42]: please all change to `Double`

File: src/main/java/com/github/obc/webserver/repositories/PlayerStatsRepository.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.github.obc.webserver.repositories;
+
+import com.github.obc.webserver.models.PlayerStats;
+import org.springframework.data.repository.CrudRepository;
+
+public interface PlayerStatsRepository extends CrudRepository<PlayerStats,String> {
```
Comment (ghost1497) [inline src/main/java/com/github/obc/webserver/repositories/PlayerStatsRepository.java:6]: Might be easier to add a `findByUserId` method

File: src/main/java/com/github/obc/webserver/controllers/PlayerStatsController.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        SimpleServiceInterface<String, PlayerStats, PlayerStatsRepository>> {
+
+    private final PlayerStatsService playerStatsService;
+
+    public PlayerStatsController(PlayerStatsService service) {
+        super(new SimpleServiceLoggerImpl<>(service));
+        this.playerStatsService = service;
+    }
+
+    @GetMapping(value = "/get-all-stats")
```
Comment (ghost1497) [inline src/main/java/com/github/obc/webserver/controllers/PlayerStatsController.java:29]: when we are picking urls for endpoints, we dont have to put the http method in the name. We can use the following:

getAllStats "/playerstats/"
getAllStatsById "/playerstats?id="
getAllStatsByAuthId "/playerstats/{authId}" 
postStats "/playerstats/"
putStats "/playerstats/"
deactivateStats "/playerstats/deactivate"
deleteStats "/playerstats/delete"

File: src/main/java/com/github/obc/webserver/services/PlayerStatsService.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.github.obc.webserver.services;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class PlayerStatsService {
```
[the remark this answers is not in the document]
Comment (ghost1497) [reply src/main/java/com/github/obc/webserver/services/PlayerStatsService.java:6]: for `getAllStatsByAuthId` , you will have to search the `UserAccount` table by the authId to get the userid then the userId to use in your query to the stats table

File: src/main/java/com/github/obc/webserver/models/PlayerStats.java (hunk 1 above)
Comment (montanizstills) [reply src/main/java/com/github/obc/webserver/models/PlayerStats.java]: resolved.

File: src/main/java/com/github/obc/webserver/controllers/PlayerStatsController.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    public ResponseEntity<PlayerStats> getAllStats() {
+        return new ResponseEntity(playerStatsService.getAllStats(), HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/get-all-stats")
```
[the remark this answers is not in the document]
Comment (montanizstills) [reply src/main/java/com/github/obc/webserver/controllers/PlayerStatsController.java]: resolved.

File: src/main/java/com/github/obc/webserver/models/PlayerStats.java (hunk 1 above)
Comment (montanizstills) [reply src/main/java/com/github/obc/webserver/models/PlayerStats.java]: resolved.

File: src/main/java/com/github/obc/webserver/models/PlayerStats.java [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+import com.github.obc.webserver.utils.services.EntityInterface;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@Entity
+public class PlayerStats implements EntityInterface<String> {
+    @Id
+    private String id;
+    private boolean statsActive;
```
[the remark this answers is not in the document]
Comment (montanizstills) [reply src/main/java/com/github/obc/webserver/models/PlayerStats.java:12]: the implementation forces the class to abide and inherit a String id, and String getId(), should we add the fields and getters in addition ?

File: src/main/java/com/github/obc/webserver/models/PlayerStats.java [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+
+import com.github.obc.webserver.utils.services.EntityInterface;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@Entity
+public class PlayerStats implements EntityInterface<String> {
+    @Id
+    private String id;
```
[the remark this answers is not in the document]
Comment (montanizstills) [reply src/main/java/com/github/obc/webserver/models/PlayerStats.java]: resolved.

File: src/main/java/com/github/obc/webserver/models/PlayerStats.java [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class PlayerStats implements EntityInterface<Long> {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+    private Long userId;
```
Comment (Git-Leon) [inline src/main/java/com/github/obc/webserver/models/PlayerStats.java:15]: Please change the type of this field to `TeamMemberAccount`.
Please change the annotation on this field to `@OneToOne`.

Please add a `PlayerStats` to `TeamMemberAccount`.
Please add `@JsonIgnore` to `playerStats`.
Please add `@OneToOne` to `playerStats`.

File: src/main/java/com/github/obc/webserver/repositories/PlayerStatsRepository.java (hunk 3 above)
Comment (montanizstills) [reply src/main/java/com/github/obc/webserver/repositories/PlayerStatsRepository.java]: resolved.

File: src/main/java/com/github/obc/webserver/models/PlayerStats.java (hunk 9 above)
Comment (montanizstills) [reply src/main/java/com/github/obc/webserver/models/PlayerStats.java:15]: resolved.

**Pull request merged**